nixos/modules/services/misc/gitea.nix

```diff
@@ -478,6 +505,53 @@ in
       ];
     };
 
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      upstreams = {
+        "backend-gitea" = {
+          servers = { ${if (cfg.settings.server.PROTOCOL == "http+unix")
+                      then "unix:${cfg.settings.server.HTTP_ADDR}"
+                      else "127.0.0.1:${toString cfg.httpPort}"
+                    } = { fail_timeout = "0"; }; };
+        };
+      };
+      virtualHosts."${cfg.settings.server.DOMAIN}" = {
+        locations."/" = {
+          proxyPass = "http://backend-gitea/";
+          recommendedProxySettings = true;
+        };
+
+        locations."^~ /avatars/" = lib.mkIf (cfg.settings.picture.AVATAR_STORAGE_TYPE == "local") {
+          alias = "${cfg.stateDir}/${cfg.settings.picture.AVATAR_UPLOAD_PATH}/";
+          tryFiles = "$uri =404";
+
+          extraConfig = ''
+            default_type "image";
+          '' + nginxCommonHeaders;
+        };
+
+        locations."^~ /repo-avatars/" = lib.mkIf (cfg.settings.picture.REPOSITORY_AVATAR_STORAGE_TYPE == "local") {
+          alias = "${cfg.stateDir}/${cfg.settings.picture.REPOSITORY_AVATAR_UPLOAD_PATH}/";
```

Comment:
This doesn't work for me as the files are owned by gitea:gitea

Reply:
Check access user `nginx` to `avatars` folder.
```
  sudo -u nginx ls /var/lib/gitea/data/avatars
```
Probably need to restart the `nginx` service so that it update the `/etc/group` file.

Reply:
the problem is that the files are chowned gitea:gitea. I restarted nginx 

```
 ➜ sudo ls -lah /var/lib/gitea/data/avatars
total 169K
drwxr-x---  3 gitea gitea   33 Oct 31 03:54 .
drwxr-x--- 16 gitea gitea   17 Nov 20 15:31 ..
-rw-r-----  1 gitea gitea 1.9K May 20  2022 00e4e60d7ef9452422d8a1a526f43778
-rw-r-----  1 gitea gitea 1.4K May 20  2022 11
-rw-r-----  1 gitea gitea 1.7K May 20  2022 12
-rw-r-----  1 gitea gitea  571 May 20  2022 13
-rw-r-----  1 gitea gitea 1.2K May 20  2022 14
-rw-r-----  1 gitea gitea  583 May 20  2022 17
```

I just noticed that the paths are not fully correct.